### PR TITLE
Frontend userauth

### PR DESCRIPTION
**Pull request by chennisden**

### CODE REVIEW

**Comment by chennisden** [comment]
@codekansas can you look into this stuff:
```
api.interceptors.request.use(
      (config) => {
        config.headers.Authorization = `Bearer ${getLocalStorageApiKey()}`;
        return config;
      },
      (error) => {
        return Promise.reject(error);
      },
    );
```

whatever you did does not seem to work

**Comment by chennisden** [comment]
Hm, there is an issue in general. Isaac's code turns out to have *inconsistent* models, in the sense that he cooked up his own types instead of using what was in Models. So the form is actually incomplete.

**Comment by chennisden** [comment]
So we're more or less back where we started. The backend on its own works perfectly fine. The frontend is still buggy (albeit buggy for different reasons)...

I'm thinking we prioritize making everything *solid* before we make any more feature requests

File: frontend/src/pages/Parts.tsx [hunk 1]
```diff
@@ -52,17 +24,6 @@ const Parts = () => {
...
-  //   parts: [
-  //     {
-  //       name: "RMD X8",
-  //       owner: "MyActuator",
-  //       description: "6:1 reduction ratio motor",
-  //       part_id: "1",
-  //       images: "https://media.robolist.xyz/rmd_x8.png",
-  //     },
-  //   ],
-  // };
```
Comment (codekansas) [inline frontend/src/pages/Parts.tsx:65]: can delete

File: store/app/api/crud/robots.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    async def list_parts(self) -> list[Part]:
+        table = await self.db.Table("Parts")
+        return [Part.model_validate(part) for part in (await table.scan())["Items"]]
+
+    async def get_part(self, part_id: str) -> Part | None:
+        table = await self.db.Table("Parts")
+        part_dict = await table.get_item(Key={"part_id": part_id})
+        if "Item" not in part_dict:
+            return None
+        return Part.model_validate(part_dict["Item"])
```
Comment (codekansas) [inline store/app/api/crud/robots.py:32]: nice :)

File: store/app/api/routers/main.py [hunk 3]
```diff
@@ -1,158 +1,27 @@
...
 
-@api_router.post("/add/robot/")
-async def add_robot(robot: Robot, crud: Annotated[Crud, Depends(Crud.get)]) -> Dict[str, str]:
-    table = await crud.db.Table("Robots")
-    robot.robot_id = str(get_new_user_id())
-    await table.put_item(Item=robot.model_dump())
-    return {"message": "Robot added successfully"}
+app.include_router(api_router, prefix="/api")
+api_router.include_router(users_router, prefix="/users", tags=["users"])
+api_router.include_router(robots_router, tags=["robots"])
```
Comment (codekansas) [inline store/app/api/routers/main.py:24]: don't want to add `prefix="/robots"` here?

File: store/app/api/routers/robot.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+
+@robots_router.get("/parts")
+async def list_parts(crud: Annotated[Crud, Depends(Crud.get)]) -> List[Part]:
+    return await crud.list_parts()
+
+
+@robots_router.get("/parts/{part_id}")
+async def get_part(part_id: str, crud: Annotated[Crud, Depends(Crud.get)]) -> Part | None:
+    return await crud.get_part(part_id)
```
Comment (codekansas) [inline store/app/api/routers/robot.py:46]: seems like this should maybe go into a separate `parts` router?

**Review by codekansas — COMMENTED** [review]
looks good, we can merge if you want - only suggestion would be to separate `/api/robots` and `/api/parts` into two separate routers, that way we can also add something like `/api/datasets` later easily too

File: store/app/api/routers/main.py (hunk 3 above)
Comment (chennisden) [reply store/app/api/routers/main.py:24]: will do when separating things... but then it'll be /robots/add instead of /add/robots/

**Pull request merged**